src/main/java/com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser.java

```diff
@@ -486,11 +516,12 @@ private static Token getMissedHtmlTag(RecognitionException exception) {
         final Deque<Token> stack = new ArrayDeque<>();
         for (int i = 0; i < tokenList.size(); i++) {
             final Token token = tokenList.get(i);
-            if (token.getType() == JavadocTokenTypes.HTML_TAG_NAME
+            final int javadocTokenType = getJavadocTokenType(token.getType());
```

Comment:
mapping to JavadocTokentypes should be done in one place, right after parsing. Non of other code should be affected by this mapping nuances.

Reply:
@romani Did you read https://github.com/checkstyle/checkstyle/pull/5455#discussion_r161795115  about this area?

Reply:
@romani The right mapping happens as the parse tree is constructed. Do you want it to be done after the complete parse tree has been constructed ? That would be inefficient and would require the traversal of almost the entire tree, wouldn't it ?